src/Spago/Command/Build.purs

```diff
@@ -135,23 +139,27 @@ run opts = do
         Right _r ->
           logSuccess "Backend build succeeded."
 
-  when workspace.buildOptions.pedanticPackages do
+  let
+    pedanticPkgs = ST.run do
+      arr <- STA.new
+      ST.foreach selectedPackages \p -> do
+        let reportSrc = pedanticPackages || (fromMaybe false $ p.package.build >>= _.pedantic_packages)
+        let reportTest = pedanticPackages || (fromMaybe false $ p.package.test >>= _.pedantic_packages)
+        when (reportSrc || reportTest) do
+          void $ STA.push (Tuple p { reportSrc, reportTest }) arr
+      STA.unsafeFreeze arr
+  unless (Array.null pedanticPkgs) do
     logInfo $ "Looking for unused and undeclared transitive dependencies..."
     maybeGraph <- Graph.runGraph globs opts.pursArgs
     for_ maybeGraph \graph -> do
       env <- ask
-      case workspace.selected of
-        Just selected -> do
-          errors <- Graph.toImportErrors selected <$> runSpago (Record.union { graph, selected } env) Graph.checkImports
-          unless (Array.null errors) do
-            die' errors
-        Nothing -> do
-          -- TODO: here we could go through all the workspace packages and run the check for each
-          -- The complication is that "dependencies" includes all the dependencies for all packages
-          errors <- map Array.fold $ for (Config.getWorkspacePackages workspace.packageSet) \selected -> do
-            Graph.toImportErrors selected <$> runSpago (Record.union { graph, selected } env) Graph.checkImports
-          unless (Array.null errors) do
-            die' errors
+      -- TODO: when the entire workspace is built, 
+      -- here we could go through all the workspace packages and run the check for each
+      -- The complication is that "dependencies" includes all the dependencies for all packages
+      errors <- map Array.fold $ for pedanticPkgs \(Tuple selected options) -> do
+        Graph.toImportErrors selected options <$> runSpago (Record.union { graph, selected } env) Graph.checkImports
+      unless (Array.null errors) do
+        die' errors
```

Comment:
Here we are calling `purs graph` for each package in the project, which is slow. We should be calling it only once with all the packages/dependencies, then make sense of it after the fact.

This will require some shuffling in the Graph module, and I am doing a similar reshuffling in #1069, so might be worth getting that in first

Reply:
Not quite. If you look a few lines earlier, you'll see a single purs graph call. I reworked this code when we did the topological sort code to separate the purs graph call from the error creation for a single package code.

Reply:
Oh I see, sorry for missing that - this is all good then!